src/get-markdown-notes.js

```diff
@@ -1,27 +1,42 @@
 const fs = require("fs");
 const path = require("path");
 
+function toRegExp(value) {
+  if (typeof value === "string") {
+    return new RegExp(`^${value}$`);
+  }
+  return value;
+}
+
+const matches = (filename) => (regExp) => regExp.test(filename);
```

Comment:
`.test()` seems like the best thing to leverage here. Unfortunately it can only be called from `RegExp`s, so that's why we convert all strings to regexps